splunk/resource_splunk_admin_saml_groups.go

```diff
@@ -130,21 +132,29 @@ func getAdminSAMLGroupsConfig(d *schema.ResourceData) (adminSAMLGroupsObject *mo
 
 func getAdminSAMLGroupsByName(name string, httpResponse *http.Response) (AdminSAMLGroupsEntry *models.AdminSAMLGroupsEntry, err error) {
 	response := &models.AdminSAMLGroupsResponse{}
+	_ = json.NewDecoder(httpResponse.Body).Decode(response)
+
 	switch httpResponse.StatusCode {
 	case 200, 201:
-		_ = json.NewDecoder(httpResponse.Body).Decode(&response)
 		re := regexp.MustCompile(`(.*)`)
 		for _, entry := range response.Entry {
 			if name == re.FindStringSubmatch(entry.Name)[1] {
 				return &entry, nil
 			}
 		}
 
-	default:
-		_ = json.NewDecoder(httpResponse.Body).Decode(response)
+	case 400:
+		// Splunk returns a 400 when a SAML group mapping is not found
+		// try to catch that here
+		re := regexp.MustCompile("Unable to find a role mapping for")
+		if re.MatchString(response.Messages[0].Text) {
+			return nil, nil
+		}
+
+		// but if the error didn't match, don't assume the 400 status was just a missing resource
 		err := errors.New(response.Messages[0].Text)
-		return AdminSAMLGroupsEntry, err
+		return nil, err
 	}
 
-	return AdminSAMLGroupsEntry, nil
+	return nil, errors.New(response.Messages[0].Text)
```

Comment:
This was previously unreachable due to there being a `return` under the `default` case, so I removed the `default` case and made the "nothing matched" return valid for a failure.